source/features/forked-to.tsx

```diff
@@ -0,0 +1,154 @@
+import './more-dropdown.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import cache from '../libs/cache';
+import {getRepoURL, getUsername} from '../libs/utils';
+import {isOwnRepo} from '../libs/page-detect';
+
+const currentRepo = getRepoURL();
+
+async function init(): Promise<void> {
+	onForkDialogOpened();
+
+	if (isOwnRepo()) {
+		onForkedPage();
+	}
+
+	await checkForks();
+}
+
+// Check for cached forks.
+async function checkForks(): Promise<void> {
+	const repo = getOriginalRepo();
+	const cached = await getCache(repo);
+	const validForks = cached.filter(validateFork);
+	for (const fork of await Promise.all(validForks)) {
+		if (fork !== currentRepo) {
+			appendHtml(fork);
+			storeCache(repo, fork);
+		}
+	}
+}
+
+// Check if the fork still exists.
+async function validateFork(repo: string): Promise<boolean> {
+	const url = new URL(`/${repo}`, location.href);
+	try {
+		const response = await fetch(String(url),
+			{
+				method: 'HEAD'
+			});
+		return response.ok;
+	} catch (error) {
+		return false;
+	}
+}
+
+// Check if we are on a forked page.
+function onForkedPage(): void {
+	const forkedFromElm = select<HTMLElement>('.fork-flag:not(.rgh-forked) a');
+	if (forkedFromElm) {
+		const forkedRepo = forkedFromElm.getAttribute('href')!.substring(1);
+		storeCache(forkedRepo, currentRepo);
+	}
+}
+
+// Check for opening the fork dialog.
+function onForkDialogOpened(): void {
+	const forkDialog = select<HTMLElement>('details-dialog[src*="/fork"]')!;
+	const forkFragment = select<HTMLElement>('include-fragment', forkDialog)!;
+	forkFragment.addEventListener('load', () => onFragmentLoaded(forkDialog));
+}
+
+// Event called when fork dialog is opened.
+function onFragmentLoaded(parent: HTMLElement): void {
+	removeAllHtml();
+
+	const repo = getOriginalRepo();
+	const forks = select.all<HTMLElement>('.octicon-repo-forked', parent).map(forkElm => {
+		const fork = forkElm.parentNode!.textContent!.trim();
+		appendHtml(fork);
+		return fork;
+	});
+
+	storeCache(repo, ...forks);
+}
+
+// Get the original repo, by checking if we are already on a fork.
+function getOriginalRepo(): string {
+	let repo = currentRepo;
+
+	const forkedFromElm = select<HTMLElement>('.fork-flag:not(.rgh-forked) a');
+	if (forkedFromElm) {
+		repo = forkedFromElm.getAttribute('href')!.substring(1);
+	}
+
+	return repo;
+}
+
+// Get cache and sort it.
+async function getCache(repo: string): Promise<string[]> {
+	const currentUser = getUsername();
+	const repoKey = key(repo);
+	const cached = await cache.get<string[]>(repoKey) || [];
+	cached.sort((a, b) => {
+		let order = a.localeCompare(b);
+		if (a.startsWith(`${currentUser}/`)) {
+			order -= 100;
```

Comment:
I don’t think that’s how it works, it should only return -1, 0, or 1. This only works because the function always returns less than 0. 

Anyway: we probably shouldn’t assume that the user’s fork has priority, we can probably avoid this check altogether.